pkg/operator/resources/batchapi/validations.go

```diff
@@ -0,0 +1,179 @@
+		}
+
+		for i, batch := range submission.ItemList.Items {
+			if len(batch) > _messageSizeLimit {
+				return ErrorItemSizeExceedsLimit(i, len(batch), _messageSizeLimit)
+			}
+		}
+
+		if submission.ItemList.BatchSize == nil {
+			submission.ItemList.BatchSize = pointer.Int(1)
+		} else if *submission.ItemList.BatchSize < 1 {
+			return errors.Wrap(cr.ErrorMustBeGreaterThanOrEqualTo(*submission.ItemList.BatchSize, 1), schema.ItemListKey, schema.BatchSizeKey)
+		}
+	}
+
+	if submission.FilePathLister != nil {
+		if submission.FilePathLister.BatchSize == nil {
+			submission.FilePathLister.BatchSize = pointer.Int(1)
+		} else if *submission.FilePathLister.BatchSize < 1 {
+			return errors.Wrap(cr.ErrorMustBeGreaterThanOrEqualTo(*submission.FilePathLister.BatchSize, 1), schema.FilePathListerKey, schema.BatchSizeKey)
+		}
+	}
+
+	if submission.DelimitedFiles != nil {
+		if submission.DelimitedFiles.BatchSize == nil {
+			submission.DelimitedFiles.BatchSize = pointer.Int(1)
+		} else if *submission.DelimitedFiles.BatchSize < 1 {
+			return errors.Wrap(cr.ErrorMustBeGreaterThanOrEqualTo(*submission.DelimitedFiles.BatchSize, 1), schema.DelimitedFilesKey, schema.BatchSizeKey)
+		}
+	}
+
+	if submission.Workers == nil {
+		return errors.Wrap(cr.ErrorCannotBeEmptyOrNull(true), schema.WorkersKey)
+	} else if *submission.Workers <= 0 {
+		return errors.Wrap(cr.ErrorMustBeGreaterThanOrEqualTo(*submission.Workers, 1), schema.WorkersKey)
+	}
+
+	return nil
+}
+
+func validateJobSubmission(submission *schema.JobSubmission) error {
+	err := jobSubmissionSchemaValidation(submission)
+	if err != nil {
+		return err
+	}
+
+	if submission.FilePathLister != nil {
+		err := validateS3Lister(&submission.DelimitedFiles.S3Lister)
+		if err != nil {
+			return errors.Wrap(err, schema.FilePathListerKey)
+		}
+	}
+
+	if submission.DelimitedFiles != nil {
+		err := validateS3Lister(&submission.DelimitedFiles.S3Lister)
+		if err != nil {
+			return errors.Wrap(err, schema.DelimitedFilesKey)
+		}
+	}
+
+	return nil
+}
+
+func validateS3Lister(s3Lister *schema.S3Lister) error {
+	if len(s3Lister.S3Paths) == 0 {
+		return errors.Wrap(cr.ErrorTooFewElements(1), schema.S3PathsKey)
+	}
+
+	for _, globPattern := range s3Lister.Includes {
+		_, err := glob.Compile(globPattern, '/')
+		if err != nil {
+			return errors.Wrap(err, schema.IncludesKey, globPattern)
+		}
+	}
+
+	for _, globPattern := range s3Lister.Excludes {
+		_, err := glob.Compile(globPattern, '/')
+		if err != nil {
+			return errors.Wrap(err, schema.ExcludesKey, globPattern)
+		}
+	}
+
+	filesFound := 0
+	for _, s3Path := range s3Lister.S3Paths {
+		if !awslib.IsValidS3Path(s3Path) {
+			return awslib.ErrorInvalidS3Path(s3Path)
+		}
+
+		err := s3IteratorFromLister(*s3Lister, func(objPath string, s3Obj *s3.Object) (bool, error) {
+			filesFound++
+			return false, nil
+		})
+
+		if err != nil {
+			return errors.Wrap(err, s3Path)
+		}
+	}
+
+	if filesFound == 0 {
```

Comment:
Do you intend to check that each s3Path has files or that any do? It seems to be checking the later. If that's what you intend, we can short-circuit the loop on s3Paths. If that's not what you indented, I think the file checking should be moved to within that loop